### ISSUE TITLE
Tests against nasty shapes

### ISSUE THREAD

**Issue by davidtheclark** [mapbox/stacker-of-worlds#6]
@mcwhittemore mentioned that he thinks `turf.intersect` has faced some limitations before. He thought Polygons with inner rings touching outer rings specifically caused problems.

We should test this library against a wide variety of shapes like that.

**Comment by morganherlocker** [comment]
> @mcwhittemore mentioned that he thinks turf.intersect has faced some limitations before. He thought Polygons with inner rings touching outer rings specifically caused problems.

Correct, this would be considered an invalid polygon by turf. Specifically, geometries must follow the [simple feature specification](https://en.wikipedia.org/wiki/Simple_Features). This is implemented upstream of turf, by the [JSTS](http://bjornharrtell.github.io/jsts/) library.

**Comment by davidtheclark** [comment]
@morganherlocker Thanks for chiming in. This is good to know. Follow-up question: Do you know of a module that checks whether a geometry does follow the simple feature specification? Or do you have any other suggestions for dealing with non-simple features when they occur?

**Comment by morganherlocker** [comment]
GEOS implementation of [ST_IsValid](http://postgis.net/docs/ST_IsValid.html) or [waygu](https://github.com/mapbox/wagyu)'s internal API are probably the most robust implementations I know of, although this can be extremely finicky. In practice, complex topology operations are expensive and unreliable as a general rule, so try/catch's that gracefully handle unexpected errors or timeouts are often required.
